clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -208,7 +208,7 @@ public class VenicePushJob implements AutoCloseable {
   public static final String MAPPER_OUTPUT_DIRECTORY = "mapper.output.directory";
 
   // static names used to construct the directory and file name
-  protected static final String VALIDATE_SCHEMA_AND_BUILD_DICTIONARY_MAPPER_OUTPUT_PARENT_DIR_DEFAULT = "/tmp/venice";
+  protected static final String VALIDATE_SCHEMA_AND_BUILD_DICTIONARY_MAPPER_OUTPUT_PARENT_DIR_DEFAULT = "/tmp";
```

Comment:
Personally, I prefer `/tmp/venice` to `/tmp` since it will help put all the venice related temp files together to avoid confusion since someone would take a look at tmp folder.

Can you look into the root cause instead of fixing it this way?

If it is really challenging, and I think we should at least have `venice-` as the prefix to tell it is for Venice usage.

Reply:
It seems Hadoop doesn't do the similar permission check as Linux:
https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HdfsPermissionsGuide.html#Overview
```
[2] Any operation that checks WRITE permission on the parent directory also checks ownership if the [sticky bit](https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HdfsPermissionsGuide.html#Overview) is set.
```
Then having `venice-` prefix in the folder created by VPJ should be good enough.

Reply:
Turns out, we also have the same issue in [Repush with TTL](https://github.com/linkedin/venice/blob/main/clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java#L383). @m-nagarajan Can we also fix that in this PR?

Reply:
Creating `/tmp/veniceMapperOutput/`, but setting the permissions only during creating the directory for the first time. Thats the template code at all the places for creating directories, but I tried to be more lenient around this leading to this issue in hdfs. Thanks for digging deeper @gaojieliu and @nisargthakkar.